src/lib.rs

```diff
@@ -1,4 +1,99 @@
 mod client;
 mod util;
+use futures::{future, stream::Stream, Async, Future, Poll};
+use http::{Request, Response};
+use hyper::{
+    client::{HttpConnector, ResponseFuture},
+    Body, Client as HyperClient, body::Payload
+};
+use bytes::Bytes;
+use tower_service::Service;
+use tower_retry::Policy;
 
 pub use self::client::{Connect, Connection};
+
+struct Client<C = HttpConnector, B = Body> {
+    inner: HyperClient<C, B>
+}
+
+impl Default for Client<HttpConnector, Body> {
+    fn default() -> Self {
+        Self {
+            inner: HyperClient::new()
+        }
+    }
+}
+
+impl <B>Service<Request<B>> for Client<B>
+where
+    Body: From<B>,
+    B: hyper::client::connect::Connect,
+    B: Payload + Send + Sync + 'static,
+    B::Data: Send,
+{
+    type Response = Response<Body>;
+    type Error = hyper::Error;
+    type Future = ResponseFuture;
+
+    fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+        Ok(Async::Ready(()))
+    }
+
+    fn call(&mut self, req: Request<B>) -> Self::Future {
+        let (parts, body) = req.into_parts();
+        let req = Request::from_parts(parts, Body::from(body));
+        self.inner.request(req)
+    }
+}
+
+#[derive(Clone)]
+struct RetryPolicy {
```

Comment:
I think this should belong in the examples as this crate isnt so much about providing nice retry policies but more to just allow someone to use hyper with tower.

Reply:
not fixed yet, but can remove.